axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/ConditionConverter.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.axonserver.connector.event;
+
+import com.google.protobuf.ByteString;
+import io.axoniq.axonserver.grpc.event.dcb.ConsistencyCondition;
+import io.axoniq.axonserver.grpc.event.dcb.Criterion;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.Tag;
+import io.axoniq.axonserver.grpc.event.dcb.TagsAndNamesCriterion;
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.annotation.Internal;
+import org.axonframework.eventsourcing.eventstore.AppendCondition;
+import org.axonframework.eventsourcing.eventstore.GlobalIndexConsistencyMarker;
+import org.axonframework.eventsourcing.eventstore.SourcingCondition;
+import org.axonframework.eventstreaming.EventCriteria;
+import org.axonframework.eventstreaming.EventCriterion;
+import org.axonframework.eventstreaming.StreamingCondition;
+import org.axonframework.messaging.QualifiedName;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Utility class containing operations to convert Axon Framework's {@link SourcingCondition} and
+ * {@link StreamingCondition} into an Axon Server {@link SourceEventsRequest} and {@link StreamEventsRequest}
+ * respectively.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+@Internal
+public abstract class ConditionConverter {
+
+    /**
+     * Converts the given {@code condition} into a {@link ConsistencyCondition}.
+     * <p>
+     * The {@link AppendCondition#consistencyMarker()}} translates to the
+     * {@link ConsistencyCondition#getConsistencyMarker() consistency marker value}. The
+     * {@link AppendCondition#criteria()} are {@link EventCriteria#flatten() flattened} before being mapped to
+     * {@link Criterion}.
+     *
+     * @param condition The {@code AppendCondition} to base the {@link ConsistencyCondition} on.
+     * @return A {@code ConsistencyCondition} based on the given {@code condition}.
+     */
+    public static ConsistencyCondition convertAppendCondition(@Nonnull AppendCondition condition) {
+        Objects.requireNonNull(condition, "The append condition cannot be null.");
```

Comment:
The null-checks in this class are runtime, not configuration-time. I think just the annotation would be enough

Reply:
Ah, you're right, we'd kill those. I'll remove it right away.